### PR TITLE
arks 1.0.3 (new formula)

### PR DESCRIPTION
**Pull request by lcoombe**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

File: Formula/arks.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  url "https://github.com/bcgsc/arks/archive/1.0.3.tar.gz"
+  sha256 "036971694e36de68d1285e385804489d478e7668459cd2fffccf9f64cf0facef"
+  head "https://github.com/bcgsc/arks.git"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "boost" => :build
+  depends_on "google-sparsehash" => :build
+  depends_on "gcc" if OS.mac? # for openmp
+  depends_on "zlib" unless OS.mac?
```
Comment (sjackman) [inline Formula/arks.rb]:
```ruby
if OS.mac?
  depends_on "gcc" # for openmp
else
  depends_on "zlib"
end
```

File: Formula/arks.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  fails_with :clang # needs openmp
+
+  def install
+    inreplace "configure.ac", "AM_INIT_AUTOMAKE", "AM_INIT_AUTOMAKE(foreign)"
+    inreplace "autogen.sh", "automake -a", "automake -ac"
+    system "./autogen.sh"
+    system "./configure", "--disable-dependency-tracking",
```
Comment (sjackman) [inline Formula/arks.rb]: I prefer a line break after the comma (and change the indenting for `brew style` after).

File: Formula/arks.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--with-boost=#{Formula["boost"].opt_include}"
+    system "make", "install"
+    prefix.install "Examples"
```
Comment (sjackman) [inline Formula/arks.rb]: You can modify `Makefile.am` so that `make install` installs these examples as well.

**Comment by sjackman** [comment]
A couple minor comments. Looks good otherwise. Thanks, Lauren.

File: Formula/arks.rb (hunk 1 above)
Comment (lcoombe) [reply Formula/arks.rb]: Done!

File: Formula/arks.rb (hunk 2 above)
Comment (lcoombe) [reply Formula/arks.rb]: Done! And fixed the white space after running `brew style arks`

File: Formula/arks.rb (hunk 3 above)
Comment (lcoombe) [reply Formula/arks.rb]: As we discussed -- I'll modify `Makefile.am` in `master`, and I changed this line to `doc.install "Examples"`

File: Formula/arks.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  def install
+    inreplace "configure.ac", "AM_INIT_AUTOMAKE", "AM_INIT_AUTOMAKE(foreign)"
+    inreplace "autogen.sh", "automake -a", "automake -ac"
+    system "./autogen.sh"
+    system "./configure",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--with-boost=#{Formula["boost"].opt_include}"
```
Comment (sjackman) [inline Formula/arks.rb]: Usually just two spaces of indent here.

**Comment by sjackman** [comment]
Thanks, Lauren!

**Pull request closed without merging**